components/ide/jetbrains/image/status/main.go

```diff
@@ -307,19 +307,65 @@ func handleSignal(projectPath string) {
 	log.Info("asked IDE to terminate")
 }
 
-func configureXmx(alias string) error {
-	xmx := os.Getenv(strings.ToUpper(alias) + "_XMX")
-	if xmx == "" {
-		return nil
+func configureVMOptions(alias string) error {
+	idePrefix := alias
+	if alias == "intellij" {
+		idePrefix = "idea"
 	}
-	launcherPath := "/ide-desktop/backend/plugins/remote-dev-server/bin/launcher.sh"
-	content, err := ioutil.ReadFile(launcherPath)
+	// [idea64|goland64|pycharm64|phpstorm64].vmoptions
+	path := fmt.Sprintf("/ide-desktop/backend/bin/%s64.vmoptions", idePrefix)
+	content, err := ioutil.ReadFile(path)
 	if err != nil {
 		return err
 	}
-	// by default remote dev already set -Xmx2048m, see /ide-desktop/backend/plugins/remote-dev-server/bin/launcher.sh
-	newContent := strings.Replace(string(content), "-Xmx2048m", "-Xmx"+xmx, 1)
-	return ioutil.WriteFile(launcherPath, []byte(newContent), 0)
+	newContent := updateVMOptions(alias, string(content))
+	return ioutil.WriteFile(path, []byte(newContent), 0)
+}
+
+// deduplicateVMOption append new VMOptions onto old VMOptions and remove any duplicated leftmost options
+func deduplicateVMOption(oldLines []string, newLines []string, predicate func(l, r string) bool) []string {
+	var result []string
+	var merged = append(oldLines, newLines...)
+	for i, left := range merged {
+		for _, right := range merged[i+1:] {
+			if predicate(left, right) {
+				left = ""
+				break
+			}
+		}
+		if left != "" {
+			result = append(result, left)
+		}
+	}
+	return result
+}
+
+func updateVMOptions(alias string, content string) string {
+	// inspired by how intellij platform merge the VMOptions
+	// https://github.com/JetBrains/intellij-community/blob/master/platform/platform-impl/src/com/intellij/openapi/application/ConfigImportHelper.java#L1115
+	filterFunc := func(l, r string) bool {
```

Comment:
why it is not part of deduplicateVMOption?

Reply:
To make it easier to have finer control over the type of VMOptions to dedup. Currently, only memory-related VMOptions (`-Xmx` / `-Xms`, etc.) are handled, however, if we need to deal with VMOptions such as GC-related by another feature toggle (e.g. a separate env var). Putting all filter logic into deduplicateVMOption is a bit messy.

It's similar to `updateVMOptions` (https://github.com/JetBrains/intellij-community/blob/master/platform/platform-impl/src/com/intellij/openapi/application/ConfigImportHelper.java#L1068) from the Intellij codebase